app/src/androidTest/java/com/example/househomey/testUtils/DatabaseSetupRule.java

```diff
@@ -42,19 +45,49 @@ public DatabaseSetupRule(Class<T> activityClass) {
     /**
      * Sets up the activity by launching it, providing user data via an intent.
      * The user data includes the user document's ID, or null if the test does not require a unique user.
-     * TODO: we can replace the "null" test user with an actual username, and prepopulate that user
-     *      in Firebase for testing non-DB updates (such as list formatting, item filtering, sorting, etc.)
+     * ESPRESSO_GENERAL_USER exists in Firebase for testing general, non-DB changing tests
+     * ex) proper page navigation, field validation, etc.
      */
     public void setupActivity() {
         Bundle userData = new Bundle();
-        userData.putString("username", isDatabaseTest ? userDoc.getId() : "null");
+        userData.putString("username", isDatabaseTest ? userDoc.getId() : "ESPRESSO_GENERAL_USER");
         ActivityScenario.launch(activityClass).onActivity(activity -> {
             Intent intent = new Intent(activity, MainActivity.class);
             intent.putExtra("userData", userData);
             activity.startActivity(intent);
         });
     }
 
+    /**
+     * Adds a mock item to the Firestore database using the provided item details.
+     * NOTE: this method is IGNORED if the test does not require/create a unique user
+     *
+     * @param itemDetails A Map<String, Object> containing the details of the mock Item to be added.
+     * @throws RuntimeException if the mock data cannot create a valid Item, if adding the mock item to Firestore
+     * fails, or if there is a timeout waiting for the operation to complete.
+     */
+    public void addMockItem(Map<String, Object> itemDetails) throws Exception {
```

Comment:
I don't think you need to change it but "mock" isn't really the right word for this, especially in light of the coding context related to "Mock". Something more like testItem or setupItem would make more sense